### DISCUSSION TITLE
Using NATS as an event store

### DISCUSSION THREAD

**Discussion by munjalpatel** [commanded/commanded#560] [Ideas]
Referencing discussion on #559 

@munjalpatel
> @slashdotdash great work!
>
> Do you think NATS would be a good alternative to Postgres for folks that need more high performance / throughput?
>
> How easy / difficult would it be to develop a NATS adapter?
> I might take a stab at it and contribute back if can get guidance :)

@yordis
> Nats aren't professionally ready for Event Sourcing; they must still add "Batch Publishing."
> 
> Developing the adapter should be easy; I have a working version of an EventStore using Nats Jetstream. I am waiting for them to add Batch Publishing.
> 
> Lastly, I would move the conversation to Discussion to keep the Issues clean 🙏🏻

**Comment by MauriceVanVeen** [comment]
Batch publishing for 2.12 will be tracked by this issue: 
https://github.com/nats-io/nats-server/issues/6549

**Reply by davydog187** [reply]
In my own testing, I found a lot of issues with NATS Jetstream, both in the client https://github.com/nats-io/nats.ex/issues/182 and with server consensus issues.

I'd personally be quite weary of using NATS as an event store until these issues were ironed out

**Reply by davydog187** [reply]
I should also note that I tried bringing these server issues to the team and got a 6 figure support quote instead.

So please be cautious

**Reply by MauriceVanVeen** [reply]
RE: server consensus issues
Should make sure you are using a server version 2.10.23+ or later, preferably 2.11.x
Consensus issues were fixed, although don't know what specifically you observed.

Can you share more about those server issues? I'm part of the server team and since the versions I mentioned it will be a lot better than prior to 2.10.23.
Not sure why you'd get such a quote? If you can point to an issue on the GitHub server repo, we're happy to help.

**Reply by davydog187** [reply]
The issue was on 2.10.22, I reached out to Synadia directly via email, so I don't have a GitHub issue.

The issue was that leader election was constantly happening for periods of time, seemingly randomly.

Glad to hear it's fixed, although we chose another solution instead.

**Reply by MauriceVanVeen** [reply]
Ah alright, very likely it's fixed by now. Although sad it came too late for your use. Thanks for sharing anyhow.
